pkg/apis/core/validation/validation_test.go

```diff
@@ -19925,6 +19926,131 @@ func TestValidateNonSpecialIP(t *testing.T) {
 	}
 }
 
+func TestValidateHostUsers(t *testing.T) {
+	falseVar := false
+	trueVar := true
+
+	cases := []struct {
+		name           string
+		success        bool
+		featureEnabled bool
+		spec           *core.PodSpec
+	}{
+		{
+			name:           "empty",
+			success:        true,
+			featureEnabled: true,
+			spec:           &core.PodSpec{},
+		},
+		{
+			name:    "f: disabled hostUsers unset",
+			success: true,
+			spec:    &core.PodSpec{},
+		},
+		{
+			name:           "f: enabled hostUsers=false",
+			featureEnabled: true,
+			success:        true,
+			spec: &core.PodSpec{
+				SecurityContext: &core.PodSecurityContext{
+					HostUsers: &falseVar,
+				},
+			},
+		},
+		{
+			name:           "f: enabled hostUsers=true",
+			featureEnabled: true,
+			success:        true,
+			spec: &core.PodSpec{
+				SecurityContext: &core.PodSecurityContext{
+					HostUsers: &trueVar,
+				},
+			},
+		},
+		{
+			name:           "f: enabled hostUsers=false & volumes",
+			featureEnabled: true,
+			success:        true,
+			spec: &core.PodSpec{
+				SecurityContext: &core.PodSecurityContext{
+					HostUsers: &trueVar,
+				},
+				Volumes: []core.Volume{
+					{
+						Name: "configmap",
+						VolumeSource: core.VolumeSource{
+							ConfigMap: &core.ConfigMapVolumeSource{
+								LocalObjectReference: core.LocalObjectReference{Name: "configmap"},
+							},
+						},
+					},
+					{
+						Name: "secret",
+						VolumeSource: core.VolumeSource{
+							Secret: &core.SecretVolumeSource{
+								SecretName: "secret",
+							},
+						},
+					},
+					{
+						Name: "downward-api",
+						VolumeSource: core.VolumeSource{
+							DownwardAPI: &core.DownwardAPIVolumeSource{},
+						},
+					},
+					{
+						Name: "proj",
+						VolumeSource: core.VolumeSource{
+							Projected: &core.ProjectedVolumeSource{},
+						},
+					},
+					{
+						Name: "empty-dir",
+						VolumeSource: core.VolumeSource{
+							EmptyDir: &core.EmptyDirVolumeSource{},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "f: enabled hostUsers=false - unsupported volume",
+			success: false,
+			spec: &core.PodSpec{
+				SecurityContext: &core.PodSecurityContext{
+					HostUsers: &trueVar,
+				},
+				Volumes: []core.Volume{
+					{
+						Name: "host-path",
+						VolumeSource: core.VolumeSource{
+							HostPath: &core.HostPathVolumeSource{},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
```

Comment:
why are we setting `tc := tc` again. I think this line is not needed

Reply:
Because go closures don't really work in this case. See: https://go.dev/doc/faq#closures_and_goroutines

It is true we are not running tests in parallel here, so it is not needed. I can remove it

Reply:
Removed, will be in the next push :)